protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/Generator.kt

```diff
@@ -310,21 +306,19 @@ class Generator : CodeGenerator {
         return functions
     }
 
-    private fun className(type: String): String {
-        val className = type.split(".").last()
-        if (compiledTypes.contains(className)) {
-            val packageName = compiledTypes[className]
-            return "$packageName.$className"
+    private fun classNameFromType(inputType: Descriptors.Descriptor): ClassName {
+        if (inputType.file.options.javaMultipleFiles) {
+            return ClassName(getFileJavaPackage(inputType.file), inputType.name)
         }
-        return type.substring(1)
+        return ClassName(getFileJavaPackage(inputType.file), getFileClassName(inputType.file), inputType.name)
```

Comment:
This isn't sufficient. It will generate an incorrect name for a message that is defined _inside of_ another message.

I was thinking of something more like so:
```suggestion
        val packageName = getFileJavaPackage(inputType.File)
        val names = getClassName(inputType).
                removePrefix(packageName).
                removePrefix(".").
                split(".").
                toTypedArray()
        return ClassName(packageName, names[0], *names.slice(1..names.size-1))
```

Reply:
You're right. I added another proto file for regression prevention